### DISCUSSION TITLE
Permettere la gestione delle segnalazioni orali e l'inserimento da parte dei riceventi

### DISCUSSION THREAD

**Discussion by frayber** [globaleaks/globaleaks-whistleblowing-software#3524] [General]
### Proposal

Dare la possibilità ai riceventi di poter inserire e gestire, lato backend, segnalazioni ricevute oralmente o in altre forme che non siano quella di Globaleaks.
In prima battuta potrebbe bastare abilitare l'inserimento di una segnalazione lato backend da parte dei riceventi, potrebbe però essere utile implementare un meccanismo di One Time Key Code così come delineato nello schema di Linee guida in materia di protezione delle persone che segnalano violazioni del diritto dell’Unione e protezione delle persone che segnalano violazioni delle disposizioni normative nazionali di ANAC.

### Motivation and context

In accordo allo "Schema di Linee guida in materia di protezione delle persone che segnalano violazioni del diritto dell’Unione e protezione delle persone che segnalano violazioni delle disposizioni normative nazionali", decreto legislativo 10 marzo 2023, n. 24, Direttiva (UE) 2019/1937 del Parlamento europeo e del Consiglio del 23 ottobre 2019, a pagina 78:
_"L'operatore, al termine dell'inserimento della segnalazione, acquisisce dalla piattaforma ANAC il codice
identificativo univoco di 16 caratteri alfanumerici (key code) di primo accesso, generato in modo casuale e
automatico dalla piattaforma informatica – detto One Time Key Code (OTKC) - che trasferisce
contestualmente (durante la telefonata) e per via orale al segnalante. Il segnalante può utilizzare il OTKC
per effettuare in autonomia il primo accesso alla segnalazione sulla piattaforma ANAC o telefonicamente
tramite operatore che accede alla segnalazione sulla piattaforma ANAC per suo conto. Al primo accesso, la
piattaforma restituisce, in cambio del OTKC, il key-code definitivo per i successivi accessi alla segnalazione."_

**Comment by evilaliv3** [comment]
Grazie @frayber;

quanto descrivi è di fatto oggi implementabile tramite dei canali nascosti.

In merito alla funzionalità OTKC si attendono le linee guida ufficiali per procedere all'implementazione.
Per tua informazione come progetto globaleaks siamo coinvolti nella progettazione delle implementazioni della futura piattaforma anac e che tutto lo sviluppo di tale piattaforma verrà continuamente fatto riconfluire in GlobaLeaks (laddove utile a tutti gli utenti e non solo all'Autorità)

I requisiti saranno sempre discussi pre-implementazione qui sul nostro ticketing system e settimanalmente vi coinvolgeremo nella discussione durante la call di sviluppo.

**Reply by frayber** [reply]
Grazie.

**Reply by mbolzan** [reply]
posso chiedere come si può gestire segnalazioni che arrivano da altri canali (telefono ecc) tramite appunto i Canali nascosti?
dove può il Ricevente inserire una segnalazione in un canale specifico (nascosto) ?

grazie

**Comment by rlovisetto** [comment]
Mi inseriscon in questo ticket per sapere se ci sono delle previsioni sulle tempistiche in merito alle implementazioni che dovranno consentire anche di poter disporre del canale "orale".
Grazie

**Comment by evilaliv3** [comment]
@rlovisetto: di fatto abbiamo già aggiunto la possibilità di inserire domande orali che permettono di registrare un messaggio vocale (https://github.com/globaleaks/GlobaLeaks/issues/3426)

L'implementazione è nota per avere ancora qualche difetto:
- manca di funzionalità per anoniizzare la registrazione, che al momento viene solamente cifrata rispetto a terze parti (https://github.com/globaleaks/GlobaLeaks/issues/3483)
- manca di compatibilità sul Tor browser dove di fatto il browser nin permette di registrare alcun vocale per ragioni di sicurezza; questa problematica non è superabile in alcun modo (https://github.com/globaleaks/GlobaLeaks/issues/3635)

**Comment by cienne1** [comment]
Buongiorno. Ci sono novità sull'implementazione dell'OTKC?
